spec/amber/cli/build_spec.cr

```diff
@@ -0,0 +1,51 @@
+{% if flag?(:run_build_tests) %}
+require "../../spec_helper"
+
+module Amber::CLI
+  begin
+    describe "building a generated app" do
+      ENV["AMBER_ENV"] = "test"
+      MainCommand.run ["new", TESTING_APP]
+      Dir.cd(TESTING_APP)
+
+      options = ["user:reference", "name:string", "body:text", "age:integer", "published:bool"]
+      MainCommand.run ["generate", "auth", "User"] | (options - ["user:reference"])
+      MainCommand.run ["generate", "scaffold", "Animal"] | options
+      MainCommand.run ["generate", "scaffold", "Post"] | options
+      MainCommand.run ["generate", "scaffold", "PostComment"] | (options + ["post:reference"])
+      MainCommand.run ["generate", "model", "Bat"] | options
+      MainCommand.run ["generate", "migration", "Crocodile"] | options
+      MainCommand.run ["generate", "mailer", "Dinosaur"] | options
+      MainCommand.run ["generate", "socket", "Eagle"] | options
+      MainCommand.run ["generate", "channel", "Falcon"] | options
```

Comment:
Any thoughts on this? I figure the best integration test for the CLI is to generate at least one of everything and make sure it still builds, the tests run, and pass.